### PR TITLE
Show details icon and tooltip on schedule and worklists

### PR DESCRIPTION
**Pull request by luketlancaster**
- [x] Add the Clubhouse Story ID: [ch24576]

### CODE REVIEW

File: src/js/views/patients/schedule/schedule-list.scss
Comment (paulfalgout) [inline src/js/views/patients/schedule/schedule-list.scss]: So _probably_ the form column should also be a px setting and the action column should be `calc(68% -32px)`.  And the headings would also need to accommodate for that.  I'm not sure what adding 16px is doing without any percentage adjustments, but something somewhere ain't quite right.

File: src/js/views/patients/shared/actions_views.js
Comment (paulfalgout) [inline src/js/views/patients/shared/actions_views.js]: So I think this could probably just be `this.$el` and then no wrapping `<span>` / `ui` is necessary

File: src/js/views/patients/worklist/worklist-list.scss
Comment (paulfalgout) [inline src/js/views/patients/worklist/worklist-list.scss:32]: Same here with injecting a fixed width inside the percentage columns.    It's possible that this should really go inside the meta since these spacing wise are really grouped together.  Makes the headings a little more interesting, but probably just a `padding-left: 24px` or something like that

**Review by paulfalgout — CHANGES_REQUESTED** [review]

File: src/js/views/patients/worklist/worklist-list.scss
Comment (LisaManresa) [reply src/js/views/patients/worklist/worklist-list.scss:32]: I'm thinking it should either live in the beginning of the meta cell or absolutely positioned on the right side of the action name cell. It would avoid wonkiness with pixels/percents in the table.

Comment (paulfalgout) [reply src/js/views/patients/worklist/worklist-list.scss:32]: I can't remember what breaks the white-space ellipsis stuff on the action. Would an absolute positioned element mess with that?

**Comment by luketlancaster** [comment]
alrighty so watcha' think about that @paulfalgout?

File: src/js/views/patients/worklist/worklist-list.scss
Comment (LisaManresa) [reply src/js/views/patients/worklist/worklist-list.scss:32]: As long as the container's padding accounts for the absolutely positioned element, putting it in the padded area, the ellipsis should work as expected.

File: src/js/views/patients/schedule/schedule-list.scss
Comment (paulfalgout) [inline src/js/views/patients/schedule/schedule-list.scss]: why would this not be `calc(68% - 32px)`?  It seems like if 16px is less than 5% then the action column width will not fill the space.

File: src/js/views/patients/worklist/worklist-list.scss
Comment (paulfalgout) [inline src/js/views/patients/worklist/worklist-list.scss:27]: Let's try:
```css
    color: #999;
    display: inline-block;
    margin-right: 16px;
    vertical-align: middle;
    width: 16px;
```

File: src/js/views/patients/worklist/action-item.hbs
Comment (paulfalgout) [inline src/js/views/patients/worklist/action-item.hbs]: can remove `table-list__meta`

File: src/js/views/patients/worklist/worklist-list.scss
Comment (paulfalgout) [inline src/js/views/patients/worklist/worklist-list.scss]: with the above requests this'd be 48px

File: src/js/views/patients/schedule/schedule-list.scss
Comment (paulfalgout) [inline src/js/views/patients/schedule/schedule-list.scss:105]: So I think this is it:
```css
.schedule-list__header-action {
  width: calc(68% - 84px);
}
.schedule-list__header-form {
  padding-left: 32px;
  width: 84px;
}
.schedule-list__action {
  width: calc(68% - 84px);
}
.schedule-list__action-details {
  color: #999;
  padding: 0 16px 0 0;
  text-align: center;
  width: 32px;
}
.schedule-list__action-form {
  padding: 0;
  width: 48px;
}
```

**Review by paulfalgout — CHANGES_REQUESTED** [review]
Not related, but can you remove the `u-margin--r-16` from the data-date-filter-region on the schedule layout.hbs?

File: src/js/views/patients/schedule/schedule-list.scss
Comment (luketlancaster) [reply src/js/views/patients/schedule/schedule-list.scss]: yeah I tried that, but couldn't get any `calc` to work at all. Just ended up filling too much space and pushing the items to the right way off the table

**Comment by luketlancaster** [comment]
ok lookin' pretty good. Only thing I saw is this: 
Screen Shot 2021-02-17 at 1 09 43 PM

the form button gets cut off when the window gets small. I think `w-40` isn't `w` enough since we keep throwing more icons into that cell

**Comment by paulfalgout** [comment]
Yeah that's a problem.  Particularly. Sept 23 at 12:00PM I believe is worst case width.. might be time for 2 custom column widths.. this one and for action name instead of the `w-`

File: src/js/views/patients/worklist/worklist-list.scss
Comment (luketlancaster) [inline src/js/views/patients/worklist/worklist-list.scss:83]: just did these two just to get them up. Sizing all adds up to 100% now so...yay?`

**Comment by luketlancaster** [comment]
alrighty this is up to date with current requests. Have at it

File: src/js/views/patients/schedule/schedule_views.js
Comment (paulfalgout) [inline src/js/views/patients/schedule/schedule_views.js]: I don't think this `const flow` is necessary.  The tooltip view isn't using it

File: src/js/views/patients/worklist/table-header.hbs
Comment (paulfalgout) [inline src/js/views/patients/worklist/table-header.hbs]: it doesn't seem like this can remain `w-40` if the column isn't.  The header should match the widths and then be adjusted

File: src/js/views/patients/shared/actions_views.js
Comment (paulfalgout) [inline src/js/views/patients/shared/actions_views.js]: I think this should check `details.length > 140`?

**Review by paulfalgout — CHANGES_REQUESTED** [review]
This is going to conflict with the test cleanup PR, but we should probably get that in first.

File: src/js/views/patients/schedule/schedule_views.js
Comment (luketlancaster) [reply src/js/views/patients/schedule/schedule_views.js]: oh yeah whoops forgot to take it out here

**Comment by luketlancaster** [comment]
sounds good. I'll wait on that PR getting merged in before I push any more changes up, but I'm ready to merge those in locally whenever

File: src/js/views/patients/shared/actions_views.js
Comment (paulfalgout) [inline src/js/views/patients/shared/actions_views.js]: I still don't understand `sliced.length` here.  `fooString.slice(0,140).length` can _never_ be greater than 140.

Let's just simplify this function and make it
```js
if (!details || details.length <= 140) return details;

return `${ details.slice(0, 140) }...`;
```

**Review by paulfalgout — CHANGES_REQUESTED** [review]
Other PR was merged. I think once we get this merge fixed with one small change, this can go in and we can run a release.

File: src/js/views/patients/shared/actions_views.js
Comment (paulfalgout) [reply src/js/views/patients/shared/actions_views.js]: Also there needs to be a test for greater than 140

File: test/integration/patients/worklist/schedule.js
Comment (paulfalgout) [inline test/integration/patients/worklist/schedule.js]: It looks like this change reduced coverage by no longer testing if a patient action links to the correct place.

**Review by paulfalgout — CHANGES_REQUESTED** [review]
coverage needs to be 100%

File: src/js/views/patients/shared/actions_views.js
Comment (paulfalgout) [reply src/js/views/patients/shared/actions_views.js]: @luketlancaster

Comment (luketlancaster) [reply src/js/views/patients/shared/actions_views.js]: ah crap missed it sorry

File: src/js/views/patients/shared/actions_views.js
Comment (paulfalgout) [inline src/js/views/patients/shared/actions_views.js]: Let's use `import trim from 'js/utils/formatting/trim';` as `trimEnd` has an alias in Edge and I don't want to deal with it or check if babel auto-fixes it etc.

**Review by paulfalgout — CHANGES_REQUESTED** [review]

File: src/js/views/patients/shared/actions_views.js
Comment (luketlancaster) [reply src/js/views/patients/shared/actions_views.js]: 👍 just pushed up so circle ci can run the tests because I don't think mine can handle a video call and cypress at the same time 🤞

**Review by paulfalgout — APPROVED** [review]

**Pull request merged**
